modules/api/php/endpoints/candidate/visit/electrophysiology/metadata.class.inc

```diff
@@ -0,0 +1,160 @@
+ *
+ * @category API
+ * @package  LORIS
+ * @author   Cecile Madjar <cecile.madjar@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+
+class Metadata extends Endpoint implements \LORIS\Middleware\ETagCalculator
+{
+    /**
+     * The requested Visit
+     *
+     * @var \Timepoint
+     */
+    private $_visit;
+
+    /**
+     * The requested Recording
+     *
+     * @var \LORIS\Recording
+     */
+    private $_recording;
+
+    /**
+     * A cache of the results of the endpoint, so that
+     * it doesn't need to be recalculated for the ETag and handler.
+     */
+    private $_cache;
+
+    /**
+     * Contructor
+     *
+     * @param \Timepoint       $visit     The requested visit
+     * @param \LORIS\Recording $recording The requested recording
+     */
+    public function __construct(\Timepoint $visit, \LORIS\Recording $recording)
+    {
+        $this->_visit     = $visit;
+        $this->_recording = $recording;
+    }
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return array('GET');
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return array(
+            'v0.0.3-dev',
+        );
+    }
+
+    /**
+     * Delegates the request to a private handler appropriate to the request's
+     * method.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+
+        case 'OPTIONS':
+            return (new \LORIS\Http\Response())
+                ->withHeader('Allow', $this->allowedMethods());
+
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Generates a response using specialized views.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        if (isset($this->_cache)) {
+            return $this->_cache;
+        }
+
+        $pathparts = $request->getAttribute('pathparts');
+        if (count($pathparts) > 1) {
+            return new \LORIS\Http\Response\JSON\NotFound();
+        }
+
+        $headername = array_shift($pathparts);
+
+        switch ($headername) {
+        case '':
```

Comment:
array_shift will return`NULL` if $pathparts is empty. I am not sure if this `case '':` will trigger.

Reply:
when I test it, I do get the full metadata header so it looks like it is triggering the correct function call, no?

Reply:
I looked it up and switch uses `==` (loose) comparison, so null is equivalent to the empty string.

Reply:
Thanks @johnsaigle